src/ucp/tag/offload/rndv.c

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright (C) 2023, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <ucp/proto/proto_single.inl>
+#include <ucp/rndv/proto_rndv.h>
+#include <ucp/rndv/proto_rndv.inl>
+#include <ucp/tag/offload.h>
+#include <ucp/tag/tag_rndv.h>
+
+
+/* Init HW rendezvous as a single protocol which does not require operation on
+ * remote side (which is the case when tag is matched - everyting is done by the
+ * HW/FW).
+ */
+static ucs_status_t
+ucp_tag_rndv_offload_proto_init(const ucp_proto_init_params_t *init_params)
+{
+    ucp_worker_h worker        = init_params->worker;
+    ucp_context_h context      = worker->context;
+    ucp_ep_config_t *ep_config = &ucs_array_elem(&worker->ep_config,
+                                                 init_params->ep_cfg_index);
+    ucp_proto_single_init_params_t params = {
+       .super.super         = *init_params,
+       .super.latency       = 0,
+       .super.overhead      = 40e-9,
+       .super.cfg_thresh    = context->config.ext.rndv_thresh,
+       .super.cfg_priority  = 60,
+       .super.min_length    = ucp_ep_tag_offload_min_rndv_thresh(ep_config),
+       .super.max_length    = ep_config->tag.offload.max_rndv_zcopy,
```

Comment:
we should take this from user config
since ep_config is result of protov1 logic calculation